### ISSUE TITLE
feat(embeddings): bidirectional Llama (LLM2Vec), Nemotron-3-Embed (Ministral 3) and LFM2.5-Embedding (non-causal short conv) on the existing backbones

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1325]
## Summary

Three embedding families are decoder backbones mlxcel already runs, re-used bidirectionally with a padding-only mask and a pooling step: bidirectional Llama (`model_type: llama_bidirec`, `architectures: ["LlamaBidirectionalModel"]`, mean pooling, the LLM2Vec recipe), Nemotron-3-Embed (`model_type: ministral3`, `architectures: ["Ministral3Model"]`, `is_causal: false`, mean pooling, Llama-4 attention scaling at offset 0) and LFM2.5-Embedding (`model_type: lfm2`, `architectures: ["Lfm2BidirectionalModel"]`, CLS pooling, non-causal short convolutions with symmetric padding). None loads today. This issue adds three wrappers over `src/models/llama3.rs`, `src/models/ministral3.rs` and `src/models/lfm2.rs`, plus the one backbone change LFM2 needs (a `conv_causal` flag on `ShortConv`).

## Current behavior

- `src/models/detection.rs:158` maps `llama` only; `llama_bidirec` is `Unsupported model type`. `Llama3Model` (`src/models/llama3.rs:950-955`) has a non-optional `lm_head` and its `forward` (line 959) ends with `self.lm_head.forward(&h)`.
- `src/models/detection.rs:283` maps `ministral3` to the causal `Ministral3Model`, whose `forward_with_caches` (`src/models/ministral3.rs:439-504`) always builds `create_causal_mask` and a sliding mask; `get_llama4_attn_scale` (line 113) is private.
- `src/models/lfm2.rs:356-386` `ShortConv::forward` always left-pads by `L_cache - 1` ("so the kernel-size-`L_cache` depthwise conv stays causal"); `forward_embeds_with_caches` (line 784) always builds `create_causal_mask` for attention layers; `Lfm2Model` fields are private (line 746-755) and there is no hidden-state output.
- All three checkpoints store keys without the `model.` prefix (`embed_tokens.weight`, `layers.{i}.*`, `norm.weight` / `embedding_norm.weight`), which every existing loader requires.

## Expected behavior

Shared: `h = embed(input_ids)`; `mask = create_bidirectional_padding_mask(attention_mask)` (`[B,1,1,L]`, f32 `0/-inf`); run every layer with a fresh, empty per-layer cache (offset 0, never reused); final norm; pool; L2-normalize. `max_length = min(sentence_bert_config max_seq_length, model_max_length, 8192)`. Prompt prefixes are caller-side and documented: all three checkpoints below declare `query: ` and `passage: ` in `config_sentence_transformers.json`.

### Bidirectional Llama

Config (verified on `nvidia/llama-nemotron-embed-1b-v2`): Llama 3.2 1B shape (`hidden_size 2048`, 16 layers, 32/8 heads, `head_dim 64`, `intermediate_size 8192`, `rope_theta 500000`, `rope_scaling {rope_type: llama3, factor 32, low_freq_factor 1, high_freq_factor 4, original_max_position_embeddings 8192}`, `rms_norm_eps 1e-5`, `tie_word_embeddings true`), plus `use_bidirectional_attention: true`, `pooling: "avg"`, `1_Pooling` mean, `max_seq_length 8192`. The existing `llama3::ModelArgs` (`src/models/llama3.rs:33`) parses all backbone keys including `rope_scaling`.

Forward: `h = embed_tokens[ids]`; `for layer: h = layer.forward(h, &mut KVCache::new(), Some(mask))` (`llama3::TransformerBlock::forward`, line 856); `h = norm(h)`; `mean_pool`; normalize.

Keys: `embed_tokens.weight`, `layers.{i}.{input_layernorm,post_attention_layernorm}.weight`, `layers.{i}.self_attn.{q,k,v,o}_proj.weight`, `layers.{i}.mlp.{gate,up,down}_proj.weight`, `norm.weight`. Sanitize: drop `rotary_emb.inv_freq`, `lm_head.*`, `position_ids`; strip a `language_model.` prefix; add `model.` when missing. LLM2Vec checkpoints that ship as PEFT adapters must be merged into a full `LlamaBidirectionalModel` checkpoint first; the loader rejects `adapter_model.safetensors`-only directories with a message saying so.

### Nemotron-3-Embed (Ministral 3 backbone)

Config (verified on `nvidia/Nemotron-3-Embed-1B-BF16` and `mlx-community/Nemotron-3-Embed-1B-BF16-8bit`): `hidden_size 2048`, 16 layers, 24/8 heads, `head_dim 128`, `intermediate_size 6144`, `rope_theta 1e6`, `rope_parameters {llama_4_scaling_beta 0.1, original_max_position_embeddings 16384, factor 16, apply_yarn_scaling false}`, `sliding_window null` (no `layer_types`, all full attention), `rms_norm_eps 1e-5`, `pad_token_id 11`, `tie_word_embeddings true`, `is_causal false`, `pooling "avg"`, `1_Pooling` mean, `max_seq_length 32768` (capped to 8192). The 8-bit conversion adds `quantization {group_size 64, bits 8, mode affine}`.

Forward:
```
h = embed_tokens[ids]
attn_scale = get_llama4_attn_scale(L, offset = 0, llama_4_scaling_beta, original_max_position_embeddings)  # [L]
for layer: h = layer.forward(h, &attn_scale, fresh cache, Some(mask))   # ministral3::TransformerBlock::forward
h = norm(h); embedding = normalize_l2(mean_pool(h, attention_mask))
```
A sliding-window layer (`layer_types` containing `sliding_attention`) would use `create_bidirectional_window_mask(attention_mask, sliding_window)`; the released checkpoints have none.

Keys: as the Llama list with `model.` missing; sanitize adds it and drops `lm_head.*`. `Ministral3Model::from_weights` (`src/models/ministral3.rs:570`) is reused as-is (its `lm_head` is `Option` and absent here).

### LFM2.5-Embedding (LFM2 backbone, non-causal conv)

Config (verified on `LiquidAI/LFM2.5-Embedding-350M`): `hidden_size 1024`, 16 layers, `layer_types` = 10 `conv` + 6 `full_attention` (indices 2, 5, 8, 10, 12, 14), `conv_L_cache 3`, `conv_bias false`, 16/8 heads (`head_dim 64`), `norm_eps 1e-5`, `vocab_size 65536`, `max_position_embeddings 128000`, `1_Pooling` `pooling_mode_cls_token: true`, `max_seq_length 512`. The tokenizer prepends `<|startoftext|>` through its post-processor; CLS pooling reads that position (right padding, so index 0).

New `ModelArgs.conv_causal: bool` (serde default `true`). When false, `ShortConv::forward` replaces the left pad of `L_cache - 1` with a symmetric pad of `L_cache / 2` on each side (`pad = 1` for `L_cache = 3`), so the output length stays `L` and each position sees `t - 1, t, t + 1`. The conv state cache is not updated in this mode (the embedder never decodes). Attention layers get `Some(mask)`; conv layers get `None`.

```
h = embed_tokens[ids]
for layer: h = layer.forward(h, fresh Lfm2LayerCache, if layer.is_attention() { Some(mask) } else { None })
h = embedding_norm(h); embedding = normalize_l2(cls_pool(h, attention_mask))
```

Keys: `embed_tokens.weight`, `embedding_norm.weight`, `layers.{i}.operator_norm.weight`, `layers.{i}.ffn_norm.weight`, `layers.{i}.feed_forward.{w1,w2,w3}.weight`, conv layers `layers.{i}.conv.{in_proj,out_proj}.weight` and `layers.{i}.conv.conv.weight [1024, 1, 3]` (transposed to `[1024, 3, 1]` by the existing sanitize at `src/models/lfm2.rs:886`), attention layers `layers.{i}.self_attn.{q,k,v,out}_proj.weight`, `layers.{i}.self_attn.{q,k}_layernorm.weight [64]`. Sanitize adds `model.` and drops `lm_head.*`.

## Implementation plan

1. `src/models/llama3.rs`: add `pub(crate) fn forward_hidden(&self, input_ids, input_embeddings: Option<&MlxArray>, caches, mask) -> [B, L, D]` (embed, layers, `norm`; no head) and make `forward_with_embeddings_impl` call it; add a `Llama3Backbone::from_weights_without_head(weights, args) -> (UnifiedEmbedding, Vec<TransformerBlock>, RMSNorm)` used when `lm_head.weight` is absent and `tie_word_embeddings` is true. `// Used by: Llama generation, LlamaBidirec, LlamaNemotronVLEmbedding`.
2. `src/models/llama_bidirec.rs` (+ `_tests.rs`): `LlamaBidirecModel { embed_tokens, layers, norm, pooling, normalize }`, `load(dir)` (args, `load_weights_from_dir_with_subfolders`, sanitize, bf16 -> f16 rule), `impl EmbeddingModel` (`default_pooling = Mean`, `embedding_dim = hidden_size`).
3. `src/models/ministral3.rs`: make `get_llama4_attn_scale` `pub(crate)`; `src/models/ministral3_embedding.rs` (+ `_tests.rs`): `Ministral3EmbeddingModel { model: Ministral3Model, pooling, normalize }` with `load(dir)` that reads `ModelArgs`, sanitizes (prefix), and calls `Ministral3Model::from_weights`; `embed` runs the loop above using `model.layers[i].forward(.., cache.as_interface(), ..)` with `model.make_caches()` per call.
4. `src/models/lfm2.rs`: `ModelArgs.conv_causal` (default true); `ShortConv { causal: bool }` with the symmetric-pad branch; `pub(crate) fn forward_hidden_bidirectional(&self, input_ids, attention_mask) -> [B, L, D]` on `Lfm2Model` that builds the padding mask and runs the loop above without touching `sequence_state`. Add `// Used by:` notes. `src/models/lfm2_embedding.rs` (+ `_tests.rs`): `Lfm2EmbeddingModel { model: Lfm2Model, pooling, normalize }`, `load(dir)` sets `args.conv_causal = false` before `Lfm2Model::from_weights`; `default_pooling = Cls`.
5. `src/models/detection.rs`: inside `is_embedding_checkpoint`: `llama_bidirec`, or `llama` with `LlamaBidirectionalModel` -> `ModelType::LlamaBidirec`; `ministral3` with `is_causal == false` or `Ministral3Model` -> `ModelType::Ministral3Embedding`; `lfm2` with `Lfm2BidirectionalModel` or a `1_Pooling` -> `ModelType::Lfm2Embedding` (a `1_Dense` folder, the late-interaction layout, is rejected with a clear message). `src/model_metadata.rs` entries; `src/embeddings/loader.rs` arms.
6. `docs/supported-models.md` rows; `docs/embeddings.md` prefixes (`query: ` / `passage: `).

## Validation

(a) Unit tests:
- `src/models/llama_bidirec_tests.rs`: `sanitize_adds_model_prefix_and_drops_head_and_inv_freq`, `bidirectional_prefill_sees_future_tokens` (96-token synthetic input, changing the last token changes position 0), `padding_invariance`, `rejects_adapter_only_directory`.
- `src/models/ministral3_embedding_tests.rs`: `attn_scale_at_offset_zero_matches_backbone` (same values as the causal path for the same `L`), `bidirectional_prefill_sees_future_tokens`, `padding_invariance`.
- `src/models/lfm2_tests.rs`: `noncausal_short_conv_symmetric_pad_keeps_length` (`L_cache 3`, a one-hot impulse at position 5 spreads to 4, 5, 6), `causal_short_conv_unchanged_when_flag_true` (byte-identical output to the pre-change path), `conv_causal_defaults_true_in_args`.
- `src/models/lfm2_embedding_tests.rs`: `cls_pooling_reads_bos_position`, `padding_invariance`.

(b) Real checkpoints:
```bash
./target/release/mlxcel download nvidia/llama-nemotron-embed-1b-v2
./target/release/mlxcel embed -m models/llama-nemotron-embed-1b-v2 \
  -p "query: how do solar panels generate electricity" \
  -p "passage: Photovoltaic cells convert sunlight into electricity through the photovoltaic effect." \
  -p "passage: The recipe calls for two cups of flour and one egg."
./target/release/mlxcel download nvidia/Nemotron-3-Embed-1B-BF16
./target/release/mlxcel embed -m models/Nemotron-3-Embed-1B-BF16 -p "query: how do solar panels generate electricity" \
  -p "passage: Photovoltaic cells convert sunlight into electricity through the photovoltaic effect." \
  -p "passage: The recipe calls for two cups of flour and one egg."
./target/release/mlxcel download LiquidAI/LFM2.5-Embedding-350M
./target/release/mlxcel embed -m models/LFM2.5-Embedding-350M -p "query: how do solar panels generate electricity" \
  -p "passage: Photovoltaic cells convert sunlight into electricity through the photovoltaic effect." \
  -p "passage: The recipe calls for two cups of flour and one egg."
./target/release/mlxcel-server -m models/Nemotron-3-Embed-1B-BF16 --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' -d '{"input": ["query: how do solar panels generate electricity", "passage: Photovoltaic cells convert sunlight into electricity through the photovoltaic effect."]}'
```
Acceptance per checkpoint: unit-norm vectors of `hidden_size` (2048, 2048, 1024); the solar passage beats the recipe passage for the query by at least 0.15; identical inputs give 1.0 within 1e-6; a batch with mixed lengths matches the single-input vectors within 1e-3 (padding and batch-order invariance); a prompt of at least 64 tokens confirms the bidirectional prefill (its position-0 hidden state differs from the same prompt truncated to 32 tokens). `mlx-community/Nemotron-3-Embed-1B-BF16-8bit` must produce vectors within cosine 0.99 of the bf16 original.

## Acceptance criteria

- [x] `LlamaBidirecModel`, `Ministral3EmbeddingModel` and `Lfm2EmbeddingModel` load their checkpoints (prefix-less keys, no `lm_head`) and run a bidirectional prefill with the padding mask
- [x] `ShortConv` supports `conv_causal = false` with symmetric padding and leaves the causal path byte-identical
- [x] the three checkpoints above pass the gates through `mlxcel embed` and `POST /v1/embeddings`
- [x] docs/supported-models.md updated
- [x] detection table in src/models/detection.rs updated with a test
- [x] cargo test --workspace --profile test-fast --features metal,accelerate passes
- [x] cargo clippy --workspace --all-targets -- -D warnings and cargo fmt --all -- --check pass

Validation host note: the port was validated on a Linux CUDA host (GB10), so the two toolchain gates ran as their CUDA equivalents, `--features cuda` in place of `--features metal,accelerate`. `metal` and `accelerate` are macOS-only and cannot be built here. The clippy and fmt gate ran as `cargo clippy --profile test-fast --features cuda --lib --bins --tests -- -D warnings` plus `cargo check --profile test-fast --features cuda --all-targets` and `cargo fmt --all -- --check`.

## Out of scope

- Nemotron-3-Embed-8B (same code).
- LFM2 / LFM2.5 ColBERT late-interaction checkpoints (`1_Dense` layout, multi-vector), candidates for a follow-up once #1337 lands.
- Merging LLM2Vec PEFT adapters inside mlxcel.

## Dependencies

Blocked by #1353

- Part of epic #1348
